oocairo.c

```
@@ -363,7 +363,10 @@ from_lua_rectangle (lua_State *L, cairo_rectangle_int_t *rect, int pos) {
 #define DO(t) \
     lua_pushstring(L, #t); \
     lua_gettable(L, pos); \
-    rect->t = luaL_checkinteger(L, -1); \
+    if (!lua_isnumber(L, -1)) { \
+        luaL_argerror(L, pos, "invalid rectangle"); \
+    } \
+    rect->t = lua_tointeger(L, -1); \
```

Hm... With Lua 5.3, `luaL_checkinteger` does more than `lua_isnumber`, because 5.5 is a number, but not an integer. But `lua_isinteger` only exists in Lua 5.3 and not in older Lua versions.

How about something like this:
```c
#if LUA_VERSION_NUM >= 503
#define oocairo_lua_isinteger(L, idx) lua_isinteger(L, idx)
#else
#define oocairo_lua_isinteger(L, idx) lua_isnumber(L, idx)
#endif
```
One could also check if the number is an integer in older Lua versions instead of silently truncating, but I am not sure if that is worth it...

---

The Lua 5.1 documentation for `luaL_checkinteger` differs from the Lua 5.3 documentation:

Lua 5.1
> luaL_checkinteger
> Checks whether the function argument narg is a number and returns this number cast to a lua_Integer. 

Lua 5.3
> luaL_checkinteger
>Checks whether the function argument arg is an integer (or can be converted to an integer) and returns this integer cast to a lua_Integer. 

So if the documentation is true there was already a silent truncation of numbers to integers before my changes.

---

I incorporated `oocairo_lua_isinteger` as you suggested, see my latest commit.